code/modules/spells/spell_types/jaunt/bloodcrawl.dm

```diff
@@ -103,10 +108,30 @@
 	blood.visible_message(span_warning("[jaunter] sinks into [blood]!"))
 	playsound(jaunt_turf, 'sound/magic/enter_blood.ogg', 50, TRUE, -1)
 	jaunter.extinguish_mob()
+	if(!iscarbon(jaunter))
```

Comment:
Make this a boolean check on the ability, not an iscarbon check.
Then make the heart grant a version with this check set to false.

Alternately, give the _demon_ a subtype of bloodcrawl which has this behaviour and the ability granted by the heart can be the base one (this might be better).

Reply:
transferred everything from ``/datum/action/cooldown/spell/jaunt/bloodcrawl`` to ``/datum/action/cooldown/spell/jaunt/bloodcrawl/slaughter_demon`` sybtype. Demon heart give parent type so now anyone who use it don't deal with bloodcrawl damage.